### PR TITLE
{Core} Delay `import`

### PR DESCRIPTION
**Pull request by jiasli**
## Description

Previously during CLI initialization, importing these modules are very time-consuming, but actually often unnecessary:

- `msrest`
- `azure.core`
- `azure.mgmt.core`
- `azure.common`

This PR delays the import until these modules are actually used. 

The effect is remarkable: For `az verion -h`, the import time dropped from 1.755s to 0.648s.
 
Before: 
image

After:
image

## Testing Guide 
```
az verion -h --verbose
```

Generate the import time graph with 
```
python -X importtime -m azure.cli version -h 2>perf.log
tuna perf.log
```

### CODE REVIEW

File: src/azure-cli-core/azure/cli/core/commands/__init__.py [hunk 1]
```diff
@@ -843,12 +845,14 @@ def _validate_cmd_level(self, ns, cmd_validator):  # pylint: disable=no-self-use
...
         for validator in getattr(ns, '_argument_validators', []):
             try:
                 validator(**self._build_kwargs(validator, ns))
-            except ValidationError:
-                logger.debug('Validation error in %s.', str(validator))
+            except Exception as ex:
+                # Delay the import and mimic an exception handler
+                from msrest.exceptions import ValidationError
+                if isinstance(ex, ValidationError):
+                    logger.debug('Validation error in %s.', str(validator))
```
Comment (jiasli) [inline src/azure-cli-core/azure/cli/core/commands/__init__.py:855]: Mimic the behavior of https://github.com/Azure/azure-cli/blob/e6e437a80639616c9871d1d9248bbb1a65be268e/src/azure-cli-core/azure/cli/core/util.py#L58-L63

File: src/azure-cli-core/azure/cli/core/commands/arm.py [hunk 2]
```diff
@@ -305,6 +304,7 @@ def assemble_json(ids):
         if full_id_list:
             setattr(namespace, '_ids', full_id_list)
 
+        from azure.mgmt.core.tools import parse_resource_id, is_valid_resource_id
```
Comment (jiasli) [inline src/azure-cli-core/azure/cli/core/commands/arm.py:307]: Migrate from `msrestazure.tools` to `azure.mgmt.core.tools`.

**Comment by haroldrandom** [comment]
what's relationship with this PR [Performance Boost](https://github.com/Azure/azure-cli/pull/13294) ?

**Comment by jiasli** [comment]
> what's relationship with this PR [Performance Boost](https://github.com/Azure/azure-cli/pull/13294) ?

This is extracted from https://github.com/Azure/azure-cli/pull/13294 to make reviewing easier.

**Pull request merged**